### PR TITLE
feat(core): use slugified IDs for brainstorm workflow handoff (#443)

### LINKED ISSUE #443: Brainstorm handoff generates overly long workflow IDs
## Problem

When a brainstorm session is handed off to implementation, the workflow `issue_id` is generated as `brainstorm-{session_uuid}`, producing IDs like `brainstorm-d9336c40-4ce9-4b12-81e1-099bb70eaa01` (47 chars). This overflows the Job Queue card in the dashboard.

Screenshot showing truncated brainstorm ID in job queue

**Source**: `amelia/server/services/brainstorm.py:980`
```python
issue_id = f"brainstorm-{session_id}"
```

## Available Context

At handoff time, we have:
- `issue_title` (user-provided, e.g., "Add dark mode support")
- `topic` (from session creation)
- `session_id` (UUID)

## Proposed Solution

Use a **slugified title + short hash** for the issue ID:

```
add-dark-mode-d9336c40
```

This provides human readability while maintaining uniqueness via the 8-char hash suffix.

### Implementation sketch

1. Add a `slugify()` utility that converts a title to a URL-safe slug (lowercase, strip special chars, replace spaces with dashes, truncate)
2. In `handoff_to_implementation()`, generate: `f"{slug}-{session_id[:8]}"` when `issue_title` is available, falling back to `f"brainstorm-{session_id[:8]}"` when it's not
3. Validate the result still passes `CreateWorkflowRequest.validate_issue_id` (alphanumeric + dashes/underscores, max 100 chars)

### Alternatives considered

| Option | Example | Trade-off |
|--------|---------|-----------|
| Short UUID only (`brainstorm-d9336c40`) | Minimal change but not human-readable |
| Title only with collision fallback | Cleanest but needs DB collision check |
| Short prefix + hash (`bs-d9336c40`) | Very short but not descriptive |

## Security Note

The current full-UUID approach is interesting from a security perspective — it makes workflow IDs unguessable. The short hash reduces entropy but since this is a local tool (not a public API), the trade-off is acceptable. The `validate_issue_id` validator already prevents injection via special characters.

## Acceptance Criteria

- [ ] Brainstorm handoff produces short, readable workflow IDs
- [ ] IDs are unique (short hash suffix)
- [ ] Fallback when no title is provided
- [ ] Existing `validate_issue_id` constraints still pass
- [ ] Dashboard job queue displays full ID without overflow

### PR DESCRIPTION
**Pull request by anderskev**
## Summary

Replaces full UUID-based workflow IDs during brainstorm handoff with short, human-readable slugified IDs derived from the issue title. This makes workflow IDs like `add-dark-mode-d9336c40` instead of `brainstorm-d9336c40-4ce9-4b12-81e1-099bb70eaa01`, improving readability in logs, CLI output, and the dashboard.

### Added
- `slugify()` utility in `amelia/core/text.py` — converts text to URL-safe slugs with configurable max length, dash-boundary truncation, and input validation
- Unit tests for `slugify()` covering truncation, edge cases, and error handling
- Integration tests for brainstorm handoff verifying short ID generation, fallback without title, and fallback for empty slugs

### Changed
- `BrainstormService.handoff_to_implementation()` now generates `{slug}-{session_id[:8]}` issue IDs instead of `brainstorm-{full_uuid}`

## Motivation

Full UUIDs in workflow IDs are hard to read and reference. Short slugified IDs give users a meaningful identifier at a glance while the 8-char session hash suffix maintains uniqueness.

Closes #443

## Testing

- [x] Unit tests added for `slugify()` (11 test cases)
- [x] Unit tests added for handoff ID generation (3 test cases)

## Checklist

- [x] Code follows project style guidelines
- [x] Self-review completed
- [x] Tests pass locally
- [x] Linting passes

---

Generated with [Claude Code](https://claude.com/claude-code)

### CODE REVIEW

File: amelia/server/services/brainstorm.py [hunk 1]
```diff
@@ -976,8 +977,9 @@ async def handoff_to_implementation(
 
         # Generate workflow ID - either from orchestrator or fallback
         if orchestrator is not None and worktree_path is not None:
-            # Create issue ID from session ID (safe characters only)
-            issue_id = f"brainstorm-{session_id}"
+            # Generate short, readable issue ID
+            slug = slugify(issue_title, max_length=15) if issue_title else ""
+            issue_id = f"{slug}-{session_id[:8]}" if slug else f"brainstorm-{session_id[:8]}"
```
[the remark this answers is not in the document]
Comment (anderskev) [reply amelia/server/services/brainstorm.py]: @coderabbitai Fixed in `808d5891`. Now falls back to `session.topic` before the generic prefix.

File: docs/plans/2026-02-15-brainstorm-workflow-id-design.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+
+### 1. `slugify()` utility
+
+New function (location TBD, likely `amelia/utils/`):
+
+- Lowercase the input
+- Replace spaces and non-alphanumeric chars with dashes
+- Collapse consecutive dashes
+- Strip leading/trailing dashes
+- Truncate to `max_length` (default 15), breaking at a dash boundary to avoid mid-word cuts
```
[the remark this answers is not in the document]
Comment (anderskev) [reply docs/plans/2026-02-15-brainstorm-workflow-id-design.md]: @coderabbitai Fixed in `808d5891`. Updated to reference `amelia/core/text.py`.
